coriolis/tests/conductor/rpc/test_server.py

```diff
@@ -18,9 +23,489 @@ def setUp(self):
+        mock_service_definition.return_value\
+            .get_available_providers.assert_called_once_with(
+                mock.sentinel.context)
+        self.assertEqual(
+            providers, mock_service_definition.return_value
+            .get_available_providers.return_value)
+
+    @mock.patch.object(server.ConductorServerEndpoint, '_scheduler_client')
+    @mock.patch.object(
+        rpc_worker_client.WorkerClient, 'from_service_definition')
+    def test_get_provider_schemas(
+        self, mock_service_definition, mock_scheduler_client):
+        schemas = self.server.get_provider_schemas(
+            mock.sentinel.context, mock.sentinel.platform_name,
+            mock.sentinel.provider_type)
+        mock_service_definition.assert_called_once_with(
+            mock_scheduler_client.get_any_worker_service(mock.sentinel.context))
+        mock_service_definition.return_value\
+            .get_provider_schemas.assert_called_once_with(
+                mock.sentinel.context, mock.sentinel.platform_name,
+                mock.sentinel.provider_type)
+        self.assertEqual(
+            schemas,
+            mock_service_definition.return_value
+            .get_provider_schemas.return_value)
+
+    @mock.patch.object(models, 'Task')
+    @mock.patch.object(uuid, 'uuid4', return_value='task_id')
+    def test_create_task(self, mock_uuid4, mock_task_model):  # pylint: disable=unused-argument
+        task1 = mock.sentinel.task1
+        task1.id = mock.sentinel.task1_id
+        task2 = mock.sentinel.task2
+        task2.id = 'task_id'
+        task2.status = constants.TASK_STATUS_SCHEDULED
+        executions = mock.sentinel.executions
+        executions.tasks = [task1, task2]
+
+        task = self.server._create_task(
+            mock.sentinel.instance, mock.sentinel.task_type, executions)
+
+        self.assertEqual(task.index, 3)
+        self.assertEqual(task.instance, mock.sentinel.instance)
+        self.assertEqual(task.status, constants.TASK_STATUS_SCHEDULED)
+
+        # Handles depends_on
+        task = self.server._create_task(
+            mock.sentinel.instance, mock.sentinel.task_type,
+            executions, depends_on=['task_id'], on_error=True)
+        self.assertEqual(task.status, constants.TASK_STATUS_SCHEDULED)
+        self.assertEqual(task.on_error, True)
+        task = self.server._create_task(
+            mock.sentinel.instance, mock.sentinel.task_type,
+            executions, depends_on=['other_task'], on_error=True)
+        self.assertEqual(task.status, constants.TASK_STATUS_ON_ERROR_ONLY)
+
+        # Handles on_error_only
+        task = self.server._create_task(
+            mock.sentinel.instance, mock.sentinel.task_type,
+            executions, depends_on=['task_id'], on_error_only=True)
+        self.assertEqual(task.on_error, True)
+        self.assertEqual(task.status, constants.TASK_STATUS_ON_ERROR_ONLY)
+
```

Comment:
Misses a case where `self.server._create_task` is called with `on_error=True`, but no `depends_on` passed.